plugins/module_utils/network/sonic/facts/lst/lst.py

```diff
@@ -0,0 +1,162 @@
+        :param connection: the device connection
+        :param ansible_facts: Facts dictionary
+        :param data: previously collected conf
+        :rtype: dictionary
+        :returns: facts
+        """
+        objs = []
+
+        if not data:
+            cfg = self.get_config(self._module)
+            data = self.update_lst(cfg)
+        objs = data
+        facts = {}
+        if objs:
+            params = utils.validate_config(self.argument_spec, {'config': objs})
+            facts['lst'] = remove_empties(params['config'])
+        ansible_facts['ansible_network_resources'].update(facts)
+        return ansible_facts
+
+    def get_config(self, module):
+        cfg = None
+        get_path = 'data/openconfig-lst-ext:lst'
+        request = {'path': get_path, 'method': 'get'}
+
+        try:
+            response = edit_config(module, to_request(module, request))
+            if 'openconfig-lst-ext:lst' in response[0][1]:
+                cfg = response[0][1].get('openconfig-lst-ext:lst')
+        except ConnectionError as exc:
+            module.fail_json(msg=str(exc), code=exc.code)
+
+        return cfg
+
+    def update_lst(self, cfg):
+        config_dict = {}
+        enum_dict = {'openconfig-lst-ext:GROUP_L3': 'l3', 'openconfig-lst-ext:ONLINE_PERCENTAGE': 'percentage'}
+
+        if cfg:
+            if 'lst-groups' in cfg and 'lst-group' in cfg['lst-groups']:
+                lst_group = cfg['lst-groups']['lst-group']
+                lst_groups_list = []
+                for group in lst_group:
+                    group_dict = {}
+                    name = group.get('name')
+                    config = group.get('config')
+
+                    if name:
+                        group_dict['name'] = name
+                    if config:
+                        all_evpn_es_downstream = config.get('all-evpn-es-downstream')
+                        all_mclags_downstream = config.get('all-mclags-downstream')
+                        group_description = config.get('description')
+                        group_type = config.get('type')
+                        threshold_down = config.get('threshold-down')
+                        threshold_type = config.get('threshold-type')
+                        threshold_up = config.get('threshold-up')
+                        timeout = config.get('timeout')
+
+                        # For all_evpn_es_downstream and all_mclags_downstream attributes, current SONiC behavior treats none value as false
+                        if all_evpn_es_downstream:
+                            group_dict['all_evpn_es_downstream'] = all_evpn_es_downstream
+                        else:
+                            group_dict['all_evpn_es_downstream'] = False
+                        if all_mclags_downstream:
+                            group_dict['all_mclags_downstream'] = all_mclags_downstream
+                        else:
+                            group_dict['all_mclags_downstream'] = False
```

Comment:
dict.get can be used to populate default values.
```suggestion
                        group_dict['all_evpn_es_downstream'] = config.get('all-evpn-es-downstream', False)
                        group_dict['all_mclags_downstream'] = config.get('all-mclags-downstream', False)
```